model_training/convert_measurements.py

```diff
@@ -27,9 +27,19 @@ def main(args: argparse.Namespace) -> None:
         save_path = Path(f"{args.result_dir}/{model_name}")
 
         layers_info = read_layers_info(engine_info_path)
-        measurements = preprocess_measurement_data(measurements_path)
-        convert_measurements_to_training_data(save_path, layers_info, measurements)
-        print(f"Saved to {save_path}!")
+        measurements = preprocess_measurement_data(
+            measurements_path, args.per_layer_measurements
+        )
+        # Create training data for a model only if all layers have
+        # sufficient samples of power and runtime measurements
+        if measurements is not None:
+            convert_measurements_to_training_data(save_path, layers_info, measurements)
+            print(f"Saved to {save_path}!")
+        else:
+            print(
```

Comment:
This doesn't feel sufficient on its own. We want to know which layers are below the threshold and by how much ideally

Reply:
This is addressed in `verify_per_layer_measurements` where we add a debug info of which layers are not sufficient and what count they have.